src/components/TodoItem.tsx

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useState } from 'react';
+import { Errors } from '../types/Error';
+
+type Props = {
+  title: string;
+  id: number;
+  completed: boolean;
+  loader: boolean;
+  deleteCurrentTodo: (id: number) => void;
+  editTodoTitle: (id: number, newTitle: string) => void;
+  setError: (error: Errors | null) => void;
+  toggleCompleted: (id: number) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  title,
+  id,
+  completed,
+  loader,
+  deleteCurrentTodo,
+  editTodoTitle,
```

Comment:
okay, but you already have title and editTitle in your props, you can use this, its looks like duplicate

Reply:
This was the function responsible for sending a request to the server and updating the todos array.
The state I created specifically for this component is editedTitle to change the input value and restore it after cancel or error.
The title properties have been extracted from the object instead of the entire object.
![Zrzut ekranu 2024-04-02 020404](https://github.com/mate-academy/react_todo-app-with-api/assets/153189782/ac3c6476-1f35-4612-83aa-77aa8f62a5a5)